secrets-provider/azure/README.md

```diff
@@ -0,0 +1,118 @@
+# Pulumi Azure KMS Encryption
+
+Pulumi allows you to use KMS encryption from your cloud provider to encrypt any secrets stored in the backend.
+
+This example shows how this might be done for Azure Keyvault. It creates a storage bucket with a single file that has a "secret" value.
+
+# Getting Started
+
+To use this example, perform the following steps. This examples assumes you have the pulumi-cli installed and the gcloud SDK installed.
+
+You should also ensure:
+
+  * You azure command line tool installed
+  * You are logging in via the `az` command line tool.
+  * You have created a resource-group
+  * You must have the environment variable `AZURE_KEYVAULT_AUTH_VIA_CLI` set to `true` eg: `export AZURE_KEYVAULT_AUTH_VIA_CLI=true`
```

Comment:
I think we should also suggest that we can do this via environment variables as we can set some for GoCloud AFAICR